server/http.go

```diff
@@ -509,31 +485,30 @@ func (p *Plugin) deauthorizeUser(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	rawInfo, appErr := p.API.KVGet(zoomTokenKeyByZoomID + req.Payload.UserID)
+	encoded, appErr := p.API.KVGet(zoomTokenKeyByZoomID + req.Payload.UserID)
 	if appErr != nil {
 		http.Error(w, appErr.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	var info ZoomUserInfo
-	err := json.Unmarshal(rawInfo, &info)
-	if err != nil {
+	var info zoom.OAuthInfo
+	if err := json.Unmarshal(encoded, &info); err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	if err = p.disconnect(info.UserID); err != nil {
+	if err := p.disconnect(info.UserID); err != nil {
 		http.Error(w, "Unable to disconnect user from Zoom", http.StatusInternalServerError)
 		return
 	}
 
-	err = p.dm(info.UserID, "We have received a deauthorization message from Zoom for your account. We have removed all your Zoom related information from our systems. Please, connect again to Zoom to keep using it.")
+	err := p.sendDirectMessage(info.UserID, "We have received a deauthorization message from Zoom for your account. We have removed all your Zoom related information from our systems. Please, connect again to Zoom to keep using it.")
 	if err != nil {
 		p.API.LogWarn("failed to dm user about deauthorization", "error", err.Error())
 	}
 
 	if req.Payload.UserDataRetention == "true" {
-		if err := p.zoomClient.CompleteCompliance(req.Payload); err != nil {
+		if err := p.apiClient.CompleteCompliance(req.Payload); err != nil {
```

Comment:
@larkox I understand that `CompleteCompliance` shouldn't necessarily be the responsibility of `JWTClient`. However, it uses the `request` method of `JWTClient` which then uses `generateJWT` again from the same client. And obviously, `generateJWT` depends on the fields of the actual `JWTClient` struct.


That's the main reason that I decided to keep it there. I'm open to suggestions though.

Reply:
We don't need to do anything fancy. It should be as simple as doing instead of `c.request` something like: `http.Post(zoomAPI + "/oauth/data/compliance", req)` (the signature of `http.Post` is not exact on my example, but something similar).

Reply:
@larkox Done: https://github.com/mattermost/mattermost-plugin-zoom/pull/150/commits/990422bc1a379e5a66decc2a3162284d905e9547